state/factory/workingset.go

```diff
@@ -338,21 +338,49 @@ func (ws *workingSet) validateNonce(ctx context.Context, blk *block.Block) error
 	if blk.Height() == 0 {
 		return nil
 	}
+	return ws.checkNonceContinuity(ctx, accountNonceMap)
+}
+
+func (ws *workingSet) validateNonce2(ctx context.Context, blk *block.Block) error {
+	accountNonceMap := make(map[string][]uint64)
+	for _, selp := range blk.Actions {
+		if action.IsSystemAction(selp) {
+			continue
+		}
+
+		caller := selp.SenderAddress()
+		if caller == nil {
+			return errors.New("failed to get address")
+		}
+		srcAddr := caller.String()
+		if _, ok := accountNonceMap[srcAddr]; !ok {
+			accountNonceMap[srcAddr] = make([]uint64, 0)
+		}
+		accountNonceMap[srcAddr] = append(accountNonceMap[srcAddr], selp.Nonce())
+	}
+
+	// Special handling for genesis block
+	if blk.Height() == 0 {
+		return nil
```

Comment:
duplicated with L338-L340, moved into checkNonceContinuity()?

Reply:
tested it's actually not needed